### PR TITLE
ROS 1 DWB Planner Code

### PR DESCRIPTION
**Pull request by crdelsey**
This an unmodified copy of most of the DWB code from https://github.com/locusrobotics/robot_navigation repo, written by @DLu 

This is intended to form the base for the ROS 2 local planner.

### CODE REVIEW

File: src/control/costmap_queue/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 2.8.3)
```
Comment (mjeronimo) [inline src/control/costmap_queue/CMakeLists.txt]: Let's push the source code down one level in the directory hierarchy. For example, under navigation2/src/control/nav2_dwb_controller. There may be opportunities for code sharing among the controllers, but we can factor that out later.

**Comment by DLu** [comment]
Just FYI: I'm currently working (in a private fork) of a new version of DWB with some ROS2 integration that will come out around ROSCon.

**Comment by crdelsey** [comment]
@DLu That's great! When you are ready to go public with it, we should discuss how we work together to integrate with the the ROS 2 nav stack changes. We were planning to make DWB the default.

**Comment by crdelsey** [comment]
This isn't even a port at this point. It is just a copy of the ROS 1 to use as a base for porting from.

**Pull request merged**